corehq/apps/reports/views.py

```diff
@@ -1131,6 +1131,7 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
         "owner_name": couch_user.full_name or couch_user.get_email(),
         "email": email,
         "notes": notes,
+        "unsub_link": get_url_base() + reverse('unsubscribe', kwargs={'user_id': owner_id}),
```

Comment:
This will take them to their account settings page where they can unsubscribe from marketing emails but not from scheduled reports.

You'll need to write a new view that removes the user from the `ReportNotification` associated with the report in question. I would guess you'd want the view to ask for confirmation before doing so and then give them a message once they have been removed.